### ISSUE TITLE
[Docs] Update logging.files.permissions documentation to consider umask

### ISSUE THREAD

**Issue by ycombinator** [elastic/beats#20584]
Based on the discussion at https://discuss.elastic.co/t/rotated-log-files-have-incorrect-permissions/244623.

After the changes done in https://github.com/elastic/beats/pull/14119 to resolve https://github.com/elastic/beats/issues/14005, the documentation about the `logging.files.permissions` setting at https://www.elastic.co/guide/en/beats/filebeat/current/configuration-logging.html#_logging_files_permissions and the example in https://www.elastic.co/guide/en/beats/filebeat/current/configuration-logging.html#configuration-logging could use an update.

**Comment by acrewdson** [comment]
Adding a +1 here since the missing info in the docs can lead to lots of confusion currently.

**Comment by kuisathaverat** [comment]
I hit this, for me is weird that you can set permissions on a parameter and those permissions will not be applied as configured, if I set `0664`, the `others` do not have permissions, and the group will not have write permission due to the creation mask.  So the last digit is ignored no matter the value, and the group will not have write permissions ever, I wonder if apply a mask make sense if you are setting explicit permissions.

**Comment by hisashinz** [comment]
A customer hit the issue.
https://elastic.my.salesforce.com/ui/support/servicedesk/ServiceDeskPage#/5004M00000i8yaa

**Comment by KeithGHarding** [comment]
I too have hit this problem.  I am using heartbeat to monitor some websites and writing the results to a file.  I have specified permissions on the output to 0604 but the "other" permissions are not being applied.  I can understand the purpose of the change was that files shouldn't be created by default to grant public permissions but the way it has been implemented is that even if I specifically state that I want public permissions this cannot be achieved. Seems like madness to me.  Rather than using a umask value (which prevents other access) wouldn't it be simpler to set the default to not grant other - then if we specify other permissions on a individual file they should be applied.
In the mean time I have seen this webpage (https://github.com/elastic/beats/pull/14119) that states:
   An individual Beat may choose to override this default umask via the libbeat/cmd/instance.Settings.Umask field.
I am not a unix expert - can someone explain how this is done???
